pkg/store/consul/consulutil/watch.go

```diff
@@ -327,7 +328,7 @@ func WatchDiff(
 				mapCopy[key] = val
 			}
 
-			outgoingChanges := &WatchedChanges{}
+			outgoingChanges := &WatchedChanges{Total: len(pairs)}
```

Comment:
saw your other PR that does this.

If we go that route we need to be sure that the caller never expects that it might be a 404. For example listening rolling updates I expect that there will be 0 during steady state. This method means EVERY caller has to protect against an empty result set but they can at least customize their behavior. will need to think about it but i think for now i prefer this approach